### PR TITLE
Manage livy endpoint from magics

### LINKED ISSUE #78: Manage livy endpoint from magics
This will be the API:

* `%spark add session_name language conn_string`
  will create a session against the endpoint specified
* `%spark info`
  will display the info for the sessions created in that notebook
* `%spark config <configuration_overrides>`
  will add session configs for subsequent sessions
* `%spark info conn_string`
  will list the sessions for a given livy endpoint by providing `session_id, language, state`
* `%spark delete session_name`
  will delete a session by its name from the notebook that created it
* `%spark delete conn_string session_id`
  will delete a session for a given endpoint by its id
* `%spark cleanup`
  will delete all sessions created by the notebook
* `%spark cleanup conn_string`
  will delete all session for the given livy endpoint

This covers #56, #75, and #76 for magics in Python kernel.
We are not designing the API for the wrapper kernels here and we'll tackle that as a separate improvement.

ping @msftristew @ellisonbg to take a look when they can

### PR DESCRIPTION
**Pull request by aggFTW**
Closes #78

### CODE REVIEW

File: remotespark/remotesparkmagics.py [hunk 1]
```diff
@@ -172,8 +193,8 @@ def _execute_against_context_that_returns_df(self, method, cell, session, output
             self.shell.write_err(e.out)
             return None
 
-    def _print_info(self):
-        sessions_info = ["\t\t{}".format(i) for i in self.spark_controller.get_manager_sessions_str()]
+    def _print_local_info(self):
+        sessions_info = ["        {}".format(i) for i in self.spark_controller.get_manager_sessions_str()]
```
Comment (msftristew) [inline remotespark/remotesparkmagics.py]: Why spaces instead of tabs?

**Comment by msftristew** [comment]
LGTM

File: remotespark/remotesparkmagics.py (hunk 1 above)
Comment (aggFTW) [inline remotespark/remotesparkmagics.py]: Because of the """ below. There's spaces there, and tabs look ridiculous.

**Comment by aggFTW** [comment]
Thanks!

**Pull request merged**